src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java

```diff
@@ -156,12 +156,25 @@ public void visitToken(DetailAST ast) {
 
     /**
      * Checks whether a method has a javadoc comment.
+     *
      * @param methodDef method definition token.
      * @return true if a method has a javadoc comment.
      */
     private static boolean hasJavadocComment(DetailAST methodDef) {
-        final DetailAST modifiers = methodDef.findFirstToken(TokenTypes.MODIFIERS);
-        return modifiers.branchContains(TokenTypes.BLOCK_COMMENT_BEGIN);
+        return tokenHasJavadocComment(methodDef, TokenTypes.MODIFIERS)
+                || tokenHasJavadocComment(methodDef, TokenTypes.TYPE);
+    }
+
+    /**
+     * Checks whether a token has a javadoc comment.
+     *
+     * @param methodDef method definition token.
+     * @param tokenType token type.
+     * @return true if a token has a javadoc comment.
+     */
+    private static boolean tokenHasJavadocComment(DetailAST methodDef, int tokenType) {
+        final DetailAST token = methodDef.findFirstToken(tokenType);
+        return token.branchContains(TokenTypes.BLOCK_COMMENT_BEGIN);
```

Comment:
This seems like another possible bug, we don't verify it is a javdoc comment just a block comment. I don't see any extra calls to `JavadocUtils.isJavadocComment` to confirm and we shouldn't be using `branchContains` anyways. See https://github.com/checkstyle/checkstyle/issues/5124 .
Block comment = `/* text */`, Javadoc comment = `/** text */`.

@RoMaNiAsHkA Please confirm my findings and create a new issue on this. This code will have to be fixed.

Reply:
Got it, I can fix all your findings in this issue, is it OK?

Reply:
I think this should be a new issue so the fixes are kept separated. It will make regression easier too.

Reply:
@rnveach, ok, I will remove blank line in javadoc and rename method in this PR, then I can create new issue for bug with block comment, based on this corrections.
I think it's more logical.